flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java

```diff
@@ -241,6 +243,19 @@
             return Optional.of("Cannot switch from job to session cluster");
         }
 
+        if (StringUtils.isNullOrWhitespaceOnly(
+                        newSpec.getFlinkConfiguration()
+                                .get(CheckpointingOptions.SAVEPOINT_DIRECTORY.key()))
+                && deployment.getStatus().getJobManagerDeploymentStatus()
```

Comment:
While it might be correct wouldn’t it be clearer to check the last reconciled job state == RUNNING?

Reply:
Imagine the current `JobManagerDeploymentStatus` is `ERROR`, and `isUpgradeModeChangedToLastStateAndHADisabledPreviously()` is true, the savepoint path also needs to be configured. Right? Because it might turn into `READY` in the following reconciliation.